packages/wallets/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useRef, useState } from 'react';
+import Calendar, { CalendarProps } from 'react-calendar';
+import CalendarIcon from '../../public/images/ic-calendar.svg';
+import FlowTextField, { TFlowFieldProps } from '../FlowField/FlowTextField';
+import unixToDateString, { customFormatShortWeekday } from './utils';
+import 'react-calendar/dist/Calendar.css';
+import './DatePicker.scss';
+
+interface TDatePickerProps extends TFlowFieldProps {
+    maxDate?: Date;
+    minDate?: Date;
+    mobileAlignment?: 'above' | 'below';
+    onDateChange: (formattedDate: string | null) => void;
+}
+
+const DatePicker = ({
+    defaultValue,
+    label,
+    maxDate,
+    message,
+    minDate,
+    mobileAlignment = 'below',
+    name,
+    onDateChange,
+    validationSchema,
+}: TDatePickerProps) => {
+    const [selectedDate, setSelectedDate] = useState<Date | null>(defaultValue ? new Date(defaultValue) : null);
+    const [isCalendarOpen, setIsCalendarOpen] = useState<boolean>(false);
+    const datePickerRef = useRef<HTMLDivElement>(null);
+
+    const toggleCalendar = () => {
+        setIsCalendarOpen(prevState => !prevState);
+    };
+
+    const handleDateChange: CalendarProps['onChange'] = value => {
+        const calendarSelectedDate = Array.isArray(value) ? value[0] : value;
+        setSelectedDate(calendarSelectedDate);
+        setIsCalendarOpen(false);
+    };
+
+    useEffect(() => {
+        if (selectedDate !== null) {
+            onDateChange(unixToDateString(selectedDate));
```

Comment:
just wondering - will it work correctly with all locales, e.g. in countries where default format is MM/DD/YYYY or DD/MM/YYYY or YYYY-MM-DD
had some funny stuff in past when handling date stored as strings as some countries do switch order between days and months and then later on date was interpreted incorrectly or even errored in certain countries,

Reply:
eventually over time we found that best practice is to store and conduct all the operations using timestamp (easy to operate, always point to specific point in time no matter the timezone or country or location) and then format it to specific format / specific timezone only in places where its needed

Reply:
https://en.wikipedia.org/wiki/Date_format_by_country

Reply:
Good question, to address your concern I updated the unixToDateString function to format the date using moment. This way the moment library would handle the date locales for us and return it based on user origin's default format